src/metabase/models/view_log.clj

```diff
@@ -18,14 +19,19 @@
   (derive ::mi/read-policy.always-allow)
   (derive ::mi/write-policy.always-allow))
 
+(mr/def ::context
+  [:maybe [:enum :dashboard :question]])
+
+(mu/defn ^:private validate
+  [_log-entry :- [:map [:context {:optional true} ::context]]]
+  :no-op)
+
 (t2/define-before-insert :model/ViewLog
   [log-entry]
+  (validate log-entry)
```

Comment:
We have `mu/validate-throw` that does something similar. Do you think we could use it here?

Reply:
Yeah! That's basically what I need, except I only want it to check the schema in dev and test mode, like with `mu/defn`. I've updated it to use the same criteria as `mu/defn` [here](https://github.com/metabase/metabase/blob/dfdaab03b3200cc21aa0ea3e4b6f8b0393090038/src/metabase/util/malli/fn.clj#L352): `(instrument-ns? *ns*)`